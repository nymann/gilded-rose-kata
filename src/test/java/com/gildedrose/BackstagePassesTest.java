package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.ClipboardReporter;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@UseReporter({ClipboardReporter.class})
public class BackstagePassesTest {

    @Test
    void qualityIncreasesBy2WhenThereAreMoreThan5ButLessThan10DaysRemaining() {
        CombinationApprovals.verifyAllCombinations(((Integer sellIn) -> {
            Item item = new Item("Backstage passes to a TAFKAL80ETC concert", sellIn, 3);
            GildedRose gildedRose = new GildedRose(new Item[]{item});

            gildedRose.updateQuality();

            return String.format("sellIn=%d, quality=%d", item.sellIn, item.quality);
        }), new Integer[]{10, 9, 8, 7, 6});
    }

    @Test
    void qualityIncreasesBy3WhenThereAre5orLessDaysRemaining() {
        CombinationApprovals.verifyAllCombinations(((Integer sellIn) -> {
            Item item = new Item("Backstage passes to a TAFKAL80ETC concert", sellIn, 20);
            GildedRose gildedRose = new GildedRose(new Item[]{item});

            gildedRose.updateQuality();

            return String.format("sellIn=%d, quality=%d", item.sellIn, item.quality);
        }), new Integer[]{5, 4, 3, 2, 1});
    }


    @Test
    void qualityDropsToZeroAfterTheConcert() {
        Item item = new Item("Backstage passes to a TAFKAL80ETC concert", 0, 10);
        GildedRose gildedRose = new GildedRose(new Item[]{item});

        gildedRose.updateQuality();
        assertEquals(0, item.quality);
    }
}
