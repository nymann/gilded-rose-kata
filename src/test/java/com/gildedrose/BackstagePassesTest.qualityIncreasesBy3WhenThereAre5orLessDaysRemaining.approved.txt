[5] => sellIn=4, quality=23 
[4] => sellIn=3, quality=23 
[3] => sellIn=2, quality=23 
[2] => sellIn=1, quality=23 
[1] => sellIn=0, quality=23 
